app/src/androidTest/java/com/github/swent/echo/compose/components/TopBarTest.kt

```diff
@@ -0,0 +1,41 @@
+package com.github.swent.echo.compose.components
+
+import androidx.compose.ui.test.junit4.createComposeRule
+import androidx.compose.ui.test.onNodeWithTag
+import androidx.compose.ui.test.onNodeWithText
+import androidx.compose.ui.test.performClick
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+
+class TopBarTest {
+
+    @get:Rule val composeTestRule = createComposeRule()
+
+    private var menuClicked = 0
+    private var searchClicked = 0
+
+    @Before
+    fun setUp() {
+        menuClicked = 0
+        searchClicked = 0
+        composeTestRule.setContent { TopBar({ menuClicked++ }, { searchClicked++ }) }
+    }
+
+    @Test
+    fun shouldShowEcho() {
+        composeTestRule.onNodeWithText("Echo").assertExists()
+    }
+
+    @Test
+    fun shouldCallOpenMenuCallbackWhenMenuButtonClicked() {
+        composeTestRule.onNodeWithTag("menu_button").performClick()
+        assert(menuClicked == 1)
```

Comment:
The  `JVM assert()` method isn't really meant to be used in test. It should be used directly in the code when some property needs to be asserted.

In tests a test framework should be preferred, that will provide more functionality, more advanced matchers and more explicit failure messages.

It will often better describe the test that you perform.

For example by using Hamcrest (which is present by default with junit in android) it could be written that way: `assertThat(menuClicked, equalTo(1))`

Same for all other `assert()` in other tests.